### PR TITLE
Improve implementation of iterators

### PR DESCRIPTION
**Pull request by weiznich**
* match the implementation of the underlying iterators

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1032,79 +1032,156 @@ pub struct Keys<'a, K: 'a, V: 'a> {
 impl<'a, K, V> Iterator for Keys<'a, K, V> {
     type Item = &'a K;
 
+    #[inline]
```
Comment (bluss) [inline src/lib.rs]: I would prefer to not specify this, it's the compiler's job. The item is already inlinable, so we don't need to do more. Same for all the other methods in the PR

File: src/lib.rs [hunk 2]
```diff
@@ -1032,79 +1032,156 @@ pub struct Keys<'a, K: 'a, V: 'a> {
...
 }
 
 impl<'a, K, V> DoubleEndedIterator for Keys<'a, K, V> {
+    #[inline]
     fn next_back(&mut self) -> Option<&'a K> {
         self.iter.next_back().map(|ent| &ent.key)
     }
 }
 
+impl<'a, K, V> ExactSizeIterator for Keys<'a, K, V> {}
```
Comment (bluss) [inline src/lib.rs]: ESI is intentionally left out, because it's not really compatible with tombstones if that's the approach that will be adopted later.

**Comment by weiznich** [comment]
Fixed

**Comment by bluss** [comment]
Thanks a lot!

**Pull request merged**